### DISCUSSION TITLE
How to modify payment types text?

### DISCUSSION THREAD

**Discussion by michaesc** [btcpayserver/btcpayserver#2943] [Technical Support]
### Q: How can a BTCPayServer administrator change the payment types in a checkout invoice?

---

![Screen shot of problem](https://raw.githubusercontent.com/michaesc/btcpayserver/master/scrshotbtcpayquestion.png)

#### Screenshot
This screenshot is taken from a public host in production.

- Prestashop 1.7.7 (PHP 7.2)
- BTCPayServer integration
- AMD64 Ubuntu 20.04 LTS
- BTCPayServer version 1.2.4
- Docker installation method

#### Problem
I don't like the way the checkout information (text and symbol) appears for the Lightening payment type.

#### Solution
Change the icon and text, **but how**?

#### Location
It seems the CSS class, SVG icon, and HTML text content is located _in sources_:

```
BTCPayServer/wwwroot/checkout/
BTCPayServer/Views/Invoice/Checkout-Body.cshtml
BTCPayServer.Common/BTCPayNetworkProvider.Bitcoin.cs
```

It seems the location of content in a Docker installation is located _in volumes_:

```
/var/lib/docker/overlay2/\<volume-id-here>/merged/app/wwwroot/
/var/lib/docker/overlay2/\<volume-id-here>/diff/app/wwwroot/
```

#### Question
How can I change the text and icon in a running BTCPayServer installation?

#### Aside
It's fine if the answer requires hard code hacking a docker volume. The more elegant solution of abstracting hard coded strings to a localised collection of text is beyond the scope of this topic.

**Comment by bolatovumar** [comment]
@michaesc I don't think there is an easy way to do this but the code that is used to make this string is here I believe: https://github.com/btcpayserver/btcpayserver/blob/f9e38deee786faf96d4ffa8a725e466d781d7b21/BTCPayServer/Payments/Lightning/LightningLikePaymentHandler.cs#L259

**Reply by michaesc** [reply]
Good answer @bolatovumar , I thnk you're right about the source location of the `Bitcoin (Lightning)` string. But this source file is not part of any installation:

```
$ find / -name '*ightningLikePaymentHandler*'
<nothing-matched>
```

So the new (corresponding) question is:

**Where does the string from my screenshot appear in a standard installation?**
---
I assume that the CS files are compiled to binary by some Microsoft or Mono compiler, something I'm totally unfamiliar with. Thanks a lot for the extra help!

**Reply by michaesc** [reply]
I finally built the project, in order to try to use the suggestion by @bolatovumar to hack the hardcoded string and rebuild to binaries.

`$ dotnet build -c Release BTCPayServer/BTCPayServer.csproj`

The new problem, is that I don't know which parts of my new build need to replace which installed (on a live server environment) components.

Running **strings(1)** on each of the binaries doesn't seem to find the text *(Lightning)* in question.

**Reply by bolatovumar** [reply]
@NicolasDorier do you know if this is possible to do?

**Reply by michaesc** [reply]
@bolatovumar @NicolasDorier it must be possible for me (or any other developer) to at least change the hard coded values and rebuild the project. The new binaries will include the custom text just like we want.

But of course the correct way to do this is to abstract all UI strings so that they can be translated or modified. In my case the text in question is simply too long for a small device and messes up the layout. That's why I'm changing it.

Once I get the time to create a full Dotnet development environment I can probably build the project with and without the text modification. After comparing the two build results, I'll try to identify the files that change.

This method is extremely inneficient, but unless we abstract the strings (maybe a long term improvement idea) it's the only way.

Do you agree?

**Reply by Marcin198** [reply]
USD Coin (USDC):

Price in USD: USD 1,01
Price in BTC: ฿ 0,00002
24h Change: -0,56%
Market cap: USD 32 803 436 893,00
Ranked: 11


Shared using: https://goo.gl/9Bwtnm
